front-end/src/helpers/AxiosWrapper.tsx

```diff
@@ -0,0 +1,34 @@
+import { Node } from '../types/Node';
+import axios from 'axios';
+
+export class AxiosWrapper {
+  static nodes: Node[] = []
+  
+  /**
+   * Given a URL, checks the host and retrieves the credentials requests to that URL should be made with.
+   * @param url to get the credentials for
+   */
+  private static getCredentialsForUrl(url: string) {
+    let nodes = AxiosWrapper.nodes.filter(n => url.includes(n.host));
+
```

Comment:
Could we log an error if no such credential was found for a given host?

Reply:
Done